openstf/tasks/optimize_hyperparameters.py

```diff
@@ -13,22 +13,68 @@
     the CRON job can be found in the /k8s/CronJobs folder.
     Alternatively this code can be run directly by running::
 
-        $ python optimize_hyper_params.py
+        $ python optimize_hyperparameters.py
 
 """
+from datetime import datetime, timedelta
+
 from openstf.pipeline.optimize_hyperparameters import optimize_hyperparameters_pipeline
 from openstf.tasks.utils.predictionjobloop import PredictionJobLoop
 from openstf.tasks.utils.taskcontext import TaskContext
+from openstf.monitoring import teams
+
+MAX_AGE_HYPER_PARAMS_DAYS = 31
+
+
+def optimize_hyperparameters_task(pj: dict, context: TaskContext) -> None:
+
+    db = context.database
+
+    # Determine if we need to optimize hyperparams
+    datetime_last_optimized = db.get_hyper_params_last_optimized(pj)
+    last_optimized_days = (datetime.utcnow() - datetime_last_optimized).days
+
+    if last_optimized_days < MAX_AGE_HYPER_PARAMS_DAYS:
+        context.logger.warning(
+            "Skip hyperparameter optimization",
+            pid=pj["id"],
+            last_optimized_days=last_optimized_days,
+            max_age=MAX_AGE_HYPER_PARAMS_DAYS
+        )
+        return
 
+    # Get input data
+    current_hyperparams = db.get_hyper_params(pj)
+    # FIXME this conversion should be done in the database
```

Comment:
This FIXME generates a code smell, better to use TODO

Reply:
But its a valid point imo. By having a smell we have a clear reminder that we need to fix this. On the other hand we don't have a clear system to use these terms `FIXME`, `TODO`, `NOTE` etc. So perhaps nicer to not have the smell and hopefully we will fix it eventually.